envoy/http/codes.h

```diff
@@ -101,7 +101,8 @@ class CodeStats {
    * routine also looks for the x-envoy-upstream-canary header and if it is set, also charges
    * canary stats.
    */
-  virtual void chargeResponseStat(const ResponseStatInfo& info) const PURE;
+  virtual void chargeResponseStat(const ResponseStatInfo& info,
+                                  bool exclude_http_code_stats = false) const PURE;
```

Comment:
Per style guide we should not user a default param here for this virtual function. I would make it required. Also, document the param above.